lib/tester.hoon

```diff
@@ -70,17 +70,30 @@
 ::
 ++  tester
   |_  eny=@uvJ
-  ::  +expect-eq: compares !>([expected actual]) and pretty-prints the result
+  ::  +expect-eq: compares :expected and :actual and pretty-prints the result
   ::
   ++  expect-eq
-    |=  a=vase
+    |=  [expected=vase actual=vase]
     ^-  tang
-    ?@  q.a  [palm+[": " ~ ~ ~]^~[>%ex-expected-pair< (sell a)]]~
-    ?:  =(-.q.a +.q.a)
-      ~
-    :~  palm+[": " ~ ~ ~]^~[leaf+"expected" (sell (slot 2 a))]
-        palm+[": " ~ ~ ~]^~[leaf+"actual" (sell (slot 3 a))]
-    ==
+    ::
+    =|  result=tang
+    ::
+    =?  result  !=(q.expected q.actual)
+      %+  weld  result
+      ^-  tang
+      :~  [%palm [": " ~ ~ ~] [leaf+"expected" (sell expected) ~]]
+          [%palm [": " ~ ~ ~] [leaf+"actual" (sell actual) ~]]
+      ==
+    ::
+    =?  result  !(~(nest ut p.actual) | p.expected)
```

Comment:
This checks whether expected nests in actual. Don't we want to check the other way? Expectations should generally be broader than specific results.

Reply:
It does depend on how we expect people to use this. If users all just write `!>(expected)`, then the type of the expected vase will generally be pretty specific, and this check makes sense. It's oddly contravariant, though.